src/java.base/share/classes/java/security/PEMEncoder.java

```diff
@@ -41,83 +39,84 @@
+ * data.  PEM is a textual encoding used to store and transfer cryptographic
  * objects, such as asymmetric keys, certificates, and certificate revocation
- * lists (CRL).  It is defined in RFC 1421 and RFC 7468.  PEM consists of a
- * Base64-formatted binary encoding enclosed by a type-identifying header
+ * lists (CRLs).  It is defined in RFC 1421 and RFC 7468.  PEM consists of a
+ * Base64-encoded binary encoding enclosed by a type-identifying header
  * and footer.
  *
- * <p> Encoding may be performed on Java API cryptographic objects that
+ * <p> Encoding can be performed on cryptographic objects that
  * implement {@link DEREncodable}. The {@link #encode(DEREncodable)}
  * and {@link #encodeToString(DEREncodable)} methods encode a DEREncodable
  * into PEM and return the data in a byte array or String.
  *
  * <p> Private keys can be encrypted and encoded by configuring a
- * {@code PEMEncoder} with the {@linkplain #withEncryption(char[])} method,
+ * {@code PEMEncoder} with the {@link #withEncryption(char[])} method,
  * which takes a password and returns a new {@code PEMEncoder} instance
  * configured to encrypt the key with that password. Alternatively, a
- * private key encrypted as an {@code EncryptedKeyInfo} object can be encoded
+ * private key encrypted as an {@link EncryptedPrivateKeyInfo} object can be encoded
  * directly to PEM by passing it to the {@code encode} or
  * {@code encodeToString} methods.
  *
- * <p> PKCS #8 2.0 defines the ASN.1 OneAsymmetricKey structure, which may
+ * <p> PKCS #8 v2.0 defines the ASN.1 OneAsymmetricKey structure, which may
  * contain both private and public keys.
  * {@link KeyPair} objects passed to the {@code encode} or
  * {@code encodeToString} methods are encoded as a
  * OneAsymmetricKey structure using the "PRIVATE KEY" type.
  *
- * <p> When encoding a {@link PEMRecord}, the API surrounds the
- * {@linkplain PEMRecord#content()} with the PEM header and footer
- * from {@linkplain PEMRecord#type()}. {@linkplain PEMRecord#leadingData()} is
- * not included in the encoding.  {@code PEMRecord} will not perform
- * validity checks on the data.
- *
- * <p>The following lists the supported {@code DEREncodable} classes and
- * the PEM types that each are encoded as:
+ * <p> When encoding a {@link PEM} object, the API surrounds
+ * {@link PEM#content()} with a PEM header and footer based on
+ * {@link PEM#type()}. The value returned by {@link PEM#leadingData()} is not
+ * included in the output.
  *
+ * <p> The following lists the supported {@code DEREncodable} classes and
+ * the PEM types they encode as:
  * <ul>
  *  <li>{@code X509Certificate} : CERTIFICATE</li>
  *  <li>{@code X509CRL} : X509 CRL</li>
- *  <li>{@code PublicKey}: PUBLIC KEY</li>
+ *  <li>{@code PublicKey} : PUBLIC KEY</li>
  *  <li>{@code PrivateKey} : PRIVATE KEY</li>
- *  <li>{@code PrivateKey} (if configured with encryption):
- *  ENCRYPTED PRIVATE KEY</li>
+ *  <li>{@code PrivateKey} :
```

Comment:
For the encodings that have multiple classes, consider combining them in one line, ex:

`PrivateKey : PRIVATE KEY or ENCRYPTED PRIVATE KEY (if configured with encryption)`